examples/jsm/geometries/ParametricGeometries.js

```diff
@@ -11,6 +11,36 @@ import { ParametricGeometry } from './ParametricGeometry.js';
 
 const ParametricGeometries = {
 
+	hopf: function ( u, v, target ) {
+
+		const pi = Math.PI;
+
+		const t = 2 * pi * u;
+		const phi = 2 * pi * v;
+
+		const a = pi / 2 - ( pi / 2 - 0.44 ) * Math.cos( 3 * t );
+		const b = t + 0.44 * Math.sin( 2 * 3 * t );
+
+		const p1 = Math.cos( a );
+		const sin_a = Math.sin( a );
+		const p2 = sin_a * Math.cos( b );
+		const p3 = sin_a * Math.sin( b );
+
+		const yden = Math.sqrt( 2 * ( 1 + p1 ) );
+		const y1 = ( 1 + p1 ) / yden;
+		const y2 = p2 / yden;
+		const y3 = p3 / yden;
+
+		const cos_phi = Math.cos( phi );
+		const sin_phi = Math.sin( phi );
+		const x1 = cos_phi * y3 + sin_phi * y2;
+		const x2 = cos_phi * y2 - sin_phi * y3;
+		const x3 = cos_phi * y1;
+		const x4 = sin_phi * y1;
+
+		target.set( x1 / ( 1 - x4 ), x2 / ( 1 - x4 ), x3 / ( 1 - x4 ) );
+	},
+
```

Comment:
This code is similar to the code from: https://laustep.github.io/stlahblog/posts/HopfTorusTennisBall.html

```js
// parameterization -------------------------------------------------
 function f0(u, v, A, n) {
	 // both u and v run from zero to one in Three.js
	 var t = n * 2 * pi * u;
	 var phi = 2 * pi * v;
	 var h = pi / 2 - (pi / 2 - A) * Math.cos(t);
	 var sinh = Math.sin(h);
	 var p2 = sinh * Math.cos(t / n + A * Math.sin(2 * t));
	 var p3 = sinh * Math.sin(t / n + A * Math.sin(2 * t));
	 var p1 = Math.cos(h);
	 var yden = Math.sqrt(2 * (1 + p1));
	 var y1 = (1 + p1) / yden;
	 var y2 = p2 / yden;
	 var y3 = p3 / yden;
	 var cosphi = Math.cos(phi);
	 var sinphi = Math.sin(phi);
	 var x4 = cosphi * y1;
	 var x3 = sinphi * y1;
	 var x2 = cosphi * y2 - sinphi * y3;
	 var x1 = cosphi * y3 + sinphi * y2;
	 return new THREE.Vector3(x1 / (1 - x4), x2 / (1 - x4), x3 / (1 - x4));
 }
 function g(A, n) {
	 return function f(u, v, vector) {
		 var out = f0(u, v, A, n);
		 vector.set(out.x, out.y, out.z);
	 }
 }
```
However, your version shows some deviations from this reference. How about adapting it 1:1 and add `https://laustep.github.io/stlahblog/posts/HopfTorusTennisBall.html` as a comment to the function.